rules/library/xcconfig.bzl

```diff
@@ -236,3 +236,93 @@ def settings_from_xcconfig(xcconfig):
         ibtool_copts = ibtool_copts,
         linkopts = linkopts,
     )
+
+def build_setting_name(build_setting):
+    """Returns the name of a given bazel build setting from the fully-qualified label
+
+    Fails if 'build_setting' is not in the expected format
+
+    Args:
+        build_setting: The fully-qualified label for a bazel build setting, e.g.,
+                       '@repo_name//path/to/package:target_name'
+    Returns:
+        The string 'target_name' in '@repo_name//path/to/package:target_name'
+    """
+    build_setting_partition = build_setting.partition(":")
+
+    if len(build_setting_partition) == 3:
+        return build_setting_partition[2]
+    else:
+        fail("Invalid build setting name: %s" % build_setting)
+
+def copts_by_build_setting_with_defaults(xcconfig = {}, fetch_default_xcconfig = {}, xcconfig_by_build_setting = {}):
+    """Creates a struct containing different configurable copts
+
+    Each returned copts is a 'select()' statement keyed by the bazel build settings in 'xcconfig_by_build_setting' and each
+    resolved value is the result of merging 'xcconfig' with the respective build setting xcconfig and applying the
+    default values from 'fetch_default_xcconfig' if necessary.
+
+    For the default value to be resolved ('//conditions:default') this macro follows the same logic described above without
+    the 'merging' step, so 'xcconfig' plus default values from 'fetch_default_xcconfig' if necessary.
+
+    Args:
+        xcconfig: A dictionary of Xcode build settings to be converted to
+                  different `copt` attributes.
+        fetch_default_xcconfig: A dictionary of default Xcode build settings
+                                to be applied for the keys that are not set.
+        xcconfig_by_build_setting: A dictionary where the keys are build settings names and
+                                   the values are the respective dictionaries of Xcode build settings
+    Returns:
+        Struct with different copts behind 'select()' statements
+    """
+    xcconfig_with_defaults = xcconfig
+
+    # Adding default values if necessary
+    for (xc_build_setting, value) in fetch_default_xcconfig.items():
+        if not xc_build_setting in xcconfig:
```

Comment:
what should we do if the setting _is_ in `xcconfig` ?

Reply:
In that case we want to ignore it unless I'm missing something 🤔 ? The default values should be applied only if they don't exist yet, otherwise `xcconfig` should be respected